README.md

```diff
@@ -1,64 +1,80 @@
 [![Build Status](https://travis-ci.com/kubevirt/common-templates.svg?branch=master)](https://travis-ci.com/kubevirt/common-templates)
 
-A set of Templates to create KubeVirt VMs.
+A set of (meta-)Templates to create KubeVirt VMs.
 
 # Overview
 
-All VMs are provided as [OpenShift templates](https://docs.okd.io/latest/dev_guide/templates.html).
-These templates can be used straight forward with OpenShift itself, or they
-can be transformed into regular objects for use with Kubernetes.
+This repository provides VM templates in the form compatible with [OpenShift templates](https://docs.okd.io/latest/dev_guide/templates.html) and kubevirt UI and those can further be transformed into regular objects for use with plain Kubernetes.
 
-Every template consists of two objects:
+The VM templates are generated from [meta-templates](templates/templates/) via [Ansible](https://www.ansible.com/) and [libosinfo](https://libosinfo.org/). The generated templates are parametrized according to three aspects: the guest OS, the workload type and the size.
 
-1. A VirtualMachineInstancePreset definition for the specific guest, this is
-   comparable to a "flavor"
-2. A VirtualMachine definition which can be used to launch the guest, if a disk
-   image is available (see below)
+The [Ansible playbook](https://docs.ansible.com/ansible/latest/user_guide/playbooks.html) [templates/generate.yaml](templates/generate.yaml) describes all combinations that should be generated.
 
-The presets allow to keep all the guest specific machine configuration in a
-single place. This configuration is applied to the VMs once they are started.
+Every template consists of a VirtualMachine definition which can be used to launch the guest, if a disk image is available (see below).
+
+# Requirements
+
+Is it necessary to install the following components to be able to run the Ansible generator and the CI suite:
+
+- jq
+- ansible >= 2.4
+- libosinfo
+- python-gobject
 
 # Usage
 
-By default the snippets below from the table will fetch the template and
-process it. It fails if a parameter (like the PVC name) is required, but not
-provided. In such a case the parameter is appended to the snippet, i.e.
-`PVCNAME`.
+By default the process below takes a generated template and converts it to an VM object that can be used to start a virtual machine.
+The conversion fails if a parameter (like the PVC name) is required, but not
+provided (i.e.`PVCNAME`).
 
 ```bash
-# Define the generic instance types:
-$ kubectl apply -f https://raw.githubusercontent.com/fabiand/common-templates/master/presets/instancetypes.yaml
+# Clone the repository
+$ git clone https://github.com/kubevirt/common-templates
+
+# Pull all submodules
+$ git submodule init update
 
-$ oc process --local -f https://git.io/fNp4Z
-The Template "win2k12r2" is invalid: template.parameters[1]: Required value:
+# Generate the template matrix
+$ pushd common-templates/templates
+$ ansible-playbook generate.yaml
```

Comment:
Maybe you can rename generate.yaml to something more specific.

I.e. generate-templates.yaml